source/common/grpc/buffered_message_ttl_manager.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <queue>
+
+#include "envoy/event/dispatcher.h"
+
+namespace Envoy {
+namespace Grpc {
+
+using BufferedMessageExpirationCallback = std::function<void(uint64_t)>;
+
+// This class is used to manage the lifetime of messages stored in BufferedAsyncClient. Messages
+// whose survival period has expired will be deleted from the Buffer.
+// We can set the ID you want to monitor TTL in the TTL Manager,
+// which will set up a callback to check for expiration.
+// You can set the ID even after the callback has been invoked.
+// When the callback is invoked, the callback given to the constructor will be
+// executed with the TTL-elapsed ID as an argument.
+// After that, if the ID to be monitored is not empty, the callback for expiration check will be set
+// again. The TTL Manager can be given a set of IDs that are expected to expire at the same time.
+// When checking for ID expiration, an expiration callback will be called for each ID
+// belonging to this set of IDs.
```

Comment:
I think it might be more natural to fold this into the start of this paragraph and talk about it handling sets of IDs + triggering callbacks for each expired ID at the beginning.

I'd also call out that there is no mechanism for preventing an expiration, so it's expected that the receiver of the callback to handle that case. 

Lastly, I'd have this comment also mention that we're using a fixed deadline per inserted entry, vs one that is configurable at insertion time